community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreNodeRelationshipCursor.java

```diff
@@ -74,24 +78,16 @@ public StoreNodeRelationshipCursor init( boolean isDense,
             int... relTypes )
     {
         this.isDense = isDense;
-        relationshipId = firstRelId;
-
+        this.relationshipId = firstRelId;
         this.fromNodeId = fromNodeId;
         this.direction = direction;
         this.relTypes = relTypes;
+        this.end = false;
 
-        if ( isDense && firstRelId != Record.NO_NEXT_RELATIONSHIP.intValue())
+        if ( isDense )
         {
-            try
-            {
-                groupRecord = relationshipGroupStore.getRecord( firstRelId, groupRecordInstance );
-                relationshipId = nextChainStart();
-            }
-            catch ( InvalidRecordException e )
-            {
-                relationshipId = Record.NO_NEXT_RELATIONSHIP.intValue();
-            }
-
+            groupStore.forceGetRecord( relationshipId, groupRecord );
+            relationshipId = nextChainStart();
```

Comment:
If the `RelationshipGroupRecord` is not in use, then the `RelationshipGroupStore.forceGetRecord` won't actually read it. Instead it'll just return an empty dummy record, where the pointers have no connection to reality. Will that be a problem for this code?